### PR TITLE
v1.4.x: update netty-tcnative version in SECURITY.md; and bump to 4.0.3

### PR DESCRIPTION
**Pull request by ejona86**
This cherry-picks #3087 to the v1.4.x branch, since updating SECURITY.md was missed before the release.

If we were strict about the versions, we would recommend 2.0.1, since that's what we were using at the time of the release, but it doesn't seem to matter much in this case and 2.0.3 is expected to be a faster and support shading.

### CODE REVIEW

**Comment by ejona86** [comment]
CC @xuwei-k

**Comment by ejona86** [comment]
@carl-mastrangelo ping

**Comment by dapengzhang0** [comment]
also `SECURITY.md` in master branch should be update to 2.05 now.

**Comment by ejona86** [comment]
No, we update SECURITY.md to what was being used at release time. We've not done any release with 2.0.5

File: SECURITY.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ In Maven, you can use the [os-maven-plugin](https://github.com/trustin/os-maven-
     <dependency>
       <groupId>io.netty</groupId>
       <artifactId>netty-tcnative-boringssl-static</artifactId>
-      <version>1.1.33.Fork26</version>
+      <version>2.0.3.Final</version>
```
Comment (carl-mastrangelo) [inline SECURITY.md]: 2.0.5

Comment (ejona86) [inline SECURITY.md]: This is a cherry-pick. Making real changes in a cherry-pick is not a great idea. I wasn't "changing it anyways", in that I didn't even open up an editor for the file.

Comment (carl-mastrangelo) [inline SECURITY.md]: Considering that there is a known bug, and we can't tell people to just use the latest version (1.5 isn't out yet), it seems reasonable to point to the latest version.  (for 1.3 and earlier it doesn't matter so much, we just tell people to upgrade)

Comment (ejona86) [inline SECURITY.md]: Then make a new commit and apply it to both master and v1.4.x. It is really important not to change cherrypicks when unnecessary and it seems unrelated to this PR.

**Comment by ejona86** [comment]
@carl-mastrangelo, are we at an impasse here?
